### DISCUSSION TITLE
CH376s SPI USB issues

### DISCUSSION THREAD

**Discussion by al789b** [djuseeq/Ch376msc#62] [Q&A]
Hi , 

I'm having some issues with wiring the chip up to my Arduino Nano/Uno following the diagram provided. I want to use SPI to read & write to a USB drive, so followed the schematic provided for SPI, but I'm sure its not detecting when the drive has been plugged in. When using the example sketch the serial monitor would return the firmware version (68) when using getChipVer(), but cannot talk to the USB drive. I doubt it is an issue with the code.

Even when I don't try to use the SPI connections but just the 5V, 3V3 and GND connections shown, I can't get the LED to light up without manually connecting it to ground with a jumper, so there may be something wrong with the wiring? Then I tried it with the other 6 CH376 modules I have and get the same issue, whether I use the Nano or Uno+logic shifter.

For the record I can't get a response using UART wiring & sketches either..

Is it likely that all of them are poorly made or am I missing something obvious? Has anyone else had a similar issue?
Any suggestions would be much appreciated. TIA :)

![CH376_SPI_SIMPLE](https://user-images.githubusercontent.com/97460751/149291156-ef63c85a-dd10-4f86-8048-d6d18c5f4e46.PNG)

![CH376_Nano_SPI_wiring](https://user-images.githubusercontent.com/97460751/149291198-fb3b0376-bf80-412b-a067-1169cfc2dd56.png)

**Comment by djuseeq** [comment]
Hi, use it with short cables, did you try with the sample code or your one? Do you use SPI only for this module? Also try the optional INTerrupt pint.

**Reply by al789b** [reply]
Hi there, thank you for the suggestions! I will have a go at using shorter cables & INT pin later today.
I am using your sample code, and ideally we want to use only SPI for this application. However, since posting I have been able to get the UART interface working & can write to a USB drive that way. 
Best regards
